apps/server/src/services/sheet-service/SheetService.ts

```diff
@@ -144,16 +143,29 @@ function verifyConnection(
     // server returns 428 if user hasnt yet completed the auth process
     try {
       logger.info(LogOrigin.Server, 'Polling for auth...');
-      const auth: Credentials = await got
-        .post(tokenUrl, {
-          json: {
-            client_id: clientSecret.installed.client_id,
-            client_secret: clientSecret.installed.client_secret,
-            device_code,
-            grant_type: grantType,
-          },
-        })
-        .json();
+
+      const response = await fetch(tokenUrl, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          client_id: clientSecret.installed.client_id,
+          client_secret: clientSecret.installed.client_secret,
+          device_code,
+          grant_type: grantType,
+        }),
+      });
+
+      if (response.status === 428) {
```

Comment:
what if: we throw as long as the response is !ok

then in the catch we can try to see the response status and send improve the error message

error message go to the logger, not the console